contracts/GnomadModule.sol

```diff
@@ -61,53 +63,49 @@ contract GnomadModule is Module {
         target = _target;
         manager = IXAppConnectionManager(_manager);
         controller = _controller;
-        origin = _origin;
+        controllerDomain = _controllerDomain;
 
         transferOwnership(_owner);
 
         emit GnomadModuleSetup(msg.sender, _owner, _avatar, _target);
     }
 
     /// @dev Check that the replica, origin, and controller are valid
-    modifier onlyValid(address _replica, uint32 _origin, bytes32 _controller) {
-        require(manager.isReplica(_replica), "sender must be a valid replica");
-        require(_origin == origin, "Unauthorized chainId");
-        require(_controller == controller, "Unauthorized controller");
+    modifier onlyValid(address _caller, uint32 _origin, bytes32 _sender) {
+        require(manager.isReplica(_caller), "caller must be a valid replica");
+        // coerce Nomad bytes32 sender to address
+        address _senderAddr = address(uint160(uint256(_sender)));
```

Comment:
https://github.com/nomad-xyz/monorepo/blob/8fa4df729900bea5a8e2471aa37fbb035ae8c1eb/packages/contracts-core/contracts/libs/TypeCasts.sol#L44

Reply:
could also import & use the Typecast library for less scary bytes conversion vibes lel